### PR TITLE
Adding penumbra Tint and a test for shadow tint and penumbra tint

### PR DESCRIPTION
**Pull request by anisunity**
All in the title

### CODE REVIEW

File: com.unity.render-pipelines.core/ShaderLibrary/CommonLighting.hlsl [hunk 1]
```diff
@@ -309,9 +309,11 @@ real ComputeMicroShadowing(real AO, real NdotL, real opacity)
 	return lerp(1.0, microshadow, opacity);
 }
 
-real3 ComputeShadowColor(real shadow, real3 shadowTint)
+real3 ComputeShadowColor(real shadow, real3 shadowTint, float penumbraFlag)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.core/ShaderLibrary/CommonLighting.hlsl]: use real not float

Comment (sebastienlagarde) [inline com.unity.render-pipelines.core/ShaderLibrary/CommonLighting.hlsl]: use real not float

File: com.unity.render-pipelines.core/ShaderLibrary/CommonLighting.hlsl [hunk 2]
```diff
@@ -309,9 +309,11 @@ real ComputeMicroShadowing(real AO, real NdotL, real opacity)
...
 {
-    return real3(1.0, 1.0, 1.0) - ((1.0 - shadow) * (real3(1.0, 1.0, 1.0) - shadowTint));
+    return lerp(real3(1.0, 1.0, 1.0) - ((1.0 - shadow) * (real3(1.0, 1.0, 1.0) - shadowTint))
+                , shadow * lerp(shadowTint, lerp(shadowTint, real3(1.0, 1.0, 1.0), shadow), shadow)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.core/ShaderLibrary/CommonLighting.hlsl]: I don't know if the compiler is able to simplify the expression, but some simplification give me
shadow * lerp(shadowTint, lerp(shadowTint, real3(1.0, 1.0, 1.0), shadow), shadow) => shadow * shadowTint + shadow ^ 3 (1 - shadowTint )   . which allow to reuse (1 - shadowTint )  with first expression. Maybe something to consider. Guess PS4 compiler is smart enough, but don't know for the others.

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs [hunk 3]
```diff
@@ -1374,13 +1381,20 @@ static Vector3 EvaluateAtmosphericAttenuation(Vector3 L, Vector3 positionWS)
...
             lightData.volumetricShadowDimmer = shadowDistanceFade * additionalLightData.volumetricShadowDimmer;
             GetContactShadowMask(additionalLightData, HDAdditionalLightData.ScalableSettings.UseContactShadow(m_Asset), hdCamera, ref lightData.contactShadowMask, ref lightData.isRayTracedContactShadow);
-            lightData.shadowTint             = new Vector3(additionalLightData.shadowTint.r, additionalLightData.shadowTint.g, additionalLightData.shadowTint.b);
+
+            // We want to have a colored penumbra if the flag is on and the color is not gray            
+            bool penumbraTint = additionalLightData.penumbraTint && ((additionalLightData.shadowTint.r != additionalLightData.shadowTint.g) || (additionalLightData.shadowTint.g != additionalLightData.shadowTint.b));
+            lightData.penumbraTint = penumbraTint ? 1.0f : 0.0f;
+            if (penumbraTint)
+                lightData.shadowTint = new Vector3(Mathf.Pow(additionalLightData.shadowTint.r, 2.2f), Mathf.Pow(additionalLightData.shadowTint.g, 2.2f), Mathf.Pow(additionalLightData.shadowTint.b, 2.2f));
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: why is there a pow 2.2 here? the pow 2.2 for conversion to sRGB ios handle by the color picker, you should have to do anything here.

**Comment by anisunity** [comment]
@JordanL8 fyi for docs

**Pull request merged**